config/v1/types.go

```diff
@@ -310,3 +310,91 @@ type DelegatedAuthorization struct {
 	// disabled indicates that authorization should be disabled.  By default it will use delegated authorization.
 	Disabled bool `json:"disabled,omitempty"`
 }
+type RequiredHSTSPolicy struct {
+	// namespaceSelector specifies a label selector such that the policy applies only to those routes that
+	// are in namespaces with labels that match the selector, and are in one of the DomainPatterns.
+	// Defaults to the empty LabelSelector, which matches everything.
+	// A domain MUST be specified in order for namespaceSelector to apply
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,1,opt,name=namespaceSelector"`
+
+	// domainPatterns is an optional list of domains for which the desired HSTS annotations are required.
+	// If domainPatterns is specified and a route is created with a spec.host matching one of the domains,
+	// the route must specify the HSTS Policy components described in the matching RequiredHSTSPolicy.
+	// If domainPatterns is empty, the specified HSTS Policy components are IGNORED.
```

Comment:
If the policy is ignored if domain patterns are empty, why would a cluster-admin want to create it?  Should we simply make the minlength=1?

Reply:
I could understand someone entering the policy, and later deleting the domainPattern as a cleanup step, but also wanting to keep it around in case a new domain wanted to use or test a proven policy.

Reply:
It seems weird to let users use the API as a note-keeping tool.  I agree with David that it makes more sense to set minlength=1.  Setting minlength=1 would also make this comment on "namespaceSelector" superfluous: "A domain MUST be specified in order for namespaceSelector to apply".

Reply:
Okay.  I will also make DomainPatterns required.